src/main/java/uk/gov/hmcts/sptribs/common/notification/CaseFinalDecisionIssuedNotification.java

```diff
@@ -105,29 +110,30 @@ private NotificationResponse sendLetterNotification(Map<String, Object> template
         return notificationService.sendLetter();
     }
 
-    private List<String> getUploadedDocumentIds(CaseData caseData) {
+    private Map<String, String> getUploadedDocuments(CaseData caseData) {
         CaseIssueFinalDecision caseIssueFinalDecision = caseData.getCaseIssueFinalDecision();
 
-        List<String> uploadedDocumentIds = new ArrayList<>();
+        Map<String, String> uploadedDocuments = new HashMap<>();
 
         String finalDecisionNotice = getFinalDecisionNoticeDocument(caseIssueFinalDecision);
-        uploadedDocumentIds.add(finalDecisionNotice);
+        uploadedDocuments.put(FINAL_DECISION_NOTICE, finalDecisionNotice);
 
         String finalDecisionGuidance = StringUtils.substringAfterLast(caseIssueFinalDecision.getFinalDecisionGuidance().getUrl(), "/");
-        uploadedDocumentIds.add(finalDecisionGuidance);
+        uploadedDocuments.put(FINAL_DECISION_GUIDANCE, finalDecisionGuidance);
 
-        return uploadedDocumentIds;
+        return uploadedDocuments;
     }
 
     private String getFinalDecisionNoticeDocument(CaseIssueFinalDecision caseIssueFinalDecision) {
-        String finalDecisionNotice;
-        if (null != caseIssueFinalDecision.getDocuments() && !caseIssueFinalDecision.getDocuments().isEmpty()) {
+        String finalDecisionNotice = null;
+        if (caseIssueFinalDecision.getFinalDecisionNotice().equals(NoticeOption.UPLOAD_FROM_COMPUTER)
+            && !CollectionUtils.isEmpty(caseIssueFinalDecision.getDocuments())) {
             List<String> uploadedDecisionNoticeDocs = caseIssueFinalDecision.getDocuments().stream().map(ListValue::getValue)
                 .map(item -> StringUtils.substringAfterLast(item.getDocumentLink().getUrl(), "/"))
                 .toList();
             finalDecisionNotice = uploadedDecisionNoticeDocs.get(0);
-
-        } else {
+        } else if (caseIssueFinalDecision.getFinalDecisionNotice().equals(NoticeOption.CREATE_FROM_TEMPLATE)
+            && null != caseIssueFinalDecision.getFinalDecisionDraft()) {
             finalDecisionNotice = StringUtils.substringAfterLast(caseIssueFinalDecision.getFinalDecisionDraft().getUrl(), "/");
         }
 
```

Comment:
Are you happy that returning a finalDecisionNotice that is null is appropriate, if neither condition is met?

Reply:
Updated default value to empty string instead of null. And that wouldnt cause any issue with template also.